diff --git a/dist/v1/Sync.js b/dist/v1/Sync.js
index 26f3e2affd6e488487e8b8491ddf1acc4b828384..79418e34ed4e375428664174a8fbe3f4164c3c70 100644
--- a/dist/v1/Sync.js
+++ b/dist/v1/Sync.js
@@ -67,9 +67,12 @@ export const makeDefaultSyncService = (config) => {
     const indexerSyncService = makeIndexerSyncService(config);
     return {
         updates: (state, secretKey) => {
-            const batchSize = 10;
-            const batchTimeout = Duration.millis(1);
-            return pipe(indexerSyncService.subscribeWallet(state), Stream.groupedWithin(batchSize, batchTimeout), Stream.map(Chunk.toArray), Stream.map((data) => WalletSyncUpdate.create(data, secretKey, new Date())), Stream.schedule(Schedule.spaced(Duration.millis(4))), Stream.provideSomeLayer(indexerSyncService.connectionLayer()));
+            const batchSize = config.batchUpdates?.size ?? 10;
+            const batchTimeout = Duration.millis(config.batchUpdates?.timeout ?? 1);
+            const batchSpacing = config.batchUpdates?.spacing ?? 4;
+            return pipe(indexerSyncService.subscribeWallet(state), Stream.groupedWithin(batchSize, batchTimeout), Stream.map(Chunk.toArray), Stream.map((data) => WalletSyncUpdate.create(data, secretKey, new Date())), batchSpacing > 0
+                ? Stream.schedule(Schedule.spaced(Duration.millis(batchSpacing)))
+                : (eventsStream) => eventsStream, Stream.provideSomeLayer(indexerSyncService.connectionLayer()));
         },
         blockData: () => {
             return Effect.gen(function* () {
